jgrapht-core/src/test/java/org/jgrapht/TestUtil.java

```diff
@@ -19,61 +19,104 @@
+  }
 
-    public static Graph<Integer, DefaultEdge> createUndirected(int[][] edges) {
-        Graph<Integer, DefaultEdge> graph = new DefaultUndirectedWeightedGraph<>(DefaultEdge.class);
-        constructGraph(graph, edges);
-        return graph;
-    }
+  public static Graph<Integer, DefaultEdge> createUndirected(int[][] edges) {
+    Graph<Integer, DefaultEdge> graph = new DefaultUndirectedWeightedGraph<>(DefaultEdge.class);
+    constructGraph(graph, edges);
+    return graph;
+  }
 
-    public static Graph<Integer, DefaultEdge> createDirected(int[][] edges) {
-        Graph<Integer, DefaultEdge> graph = new DefaultDirectedWeightedGraph<>(DefaultEdge.class);
-        constructGraph(graph, edges);
-        return graph;
-    }
+  public static Graph<Integer, DefaultEdge> createDirected(int[][] edges) {
+    Graph<Integer, DefaultEdge> graph = new DefaultDirectedWeightedGraph<>(DefaultEdge.class);
+    constructGraph(graph, edges);
+    return graph;
+  }
 
-    public static <V> Graph<V, DefaultEdge> createDirected(V[][] edges) {
-        Graph<V, DefaultEdge> graph = new DefaultDirectedWeightedGraph<>(DefaultEdge.class);
-        constructGraph(graph, edges);
-        return graph;
-    }
+  public static <V> Graph<V, DefaultEdge> createDirected(V[][] edges) {
+    Graph<V, DefaultEdge> graph = new DefaultDirectedWeightedGraph<>(DefaultEdge.class);
+    constructGraph(graph, edges);
+    return graph;
+  }
 
-    public static Graph<Integer, DefaultEdge> createPseudograph(int[][] edges){
-        Graph<Integer, DefaultEdge> graph = new WeightedPseudograph<>(DefaultEdge.class);
-        constructGraph(graph, edges);
-        return graph;
-    }
+  public static Graph<Integer, DefaultEdge> createPseudograph(int[][] edges) {
+    Graph<Integer, DefaultEdge> graph = new WeightedPseudograph<>(DefaultEdge.class);
+    constructGraph(graph, edges);
+    return graph;
+  }
 
-    public static <V> Graph<V, DefaultEdge> createPseudograph(V[][] edges){
-        Graph<V, DefaultEdge> graph = new WeightedPseudograph<>(DefaultEdge.class);
-        constructGraph(graph, edges);
-        return graph;
-    }
+  public static <V> Graph<V, DefaultEdge> createPseudograph(V[][] edges) {
+    Graph<V, DefaultEdge> graph = new WeightedPseudograph<>(DefaultEdge.class);
+    constructGraph(graph, edges);
+    return graph;
+  }
+
+  @SafeVarargs
+  public static <V, E> void addVertices(final Graph<V, E> graph, final V... vertices) {
+    for (V vertex : vertices)
+      graph.addVertex(vertex);
+  }
+
+  @SafeVarargs
+  public static <V, E> void addEdges(final Graph<V, E> graph, final V... vertices) {
+    if (vertices.length % 2 != 0)
+      throw new IllegalArgumentException("number of vertices (edge pairs) must be even");
+    for (int i = 0; i < vertices.length; i = i + 2)
+      graph.addEdge(vertices[i], vertices[i + 1]);
+  }
+
+  @SafeVarargs
+  public static <V, E> void addHamiltonianCycle(final Graph<V, E> graph, final V... vertices) {
```

Comment:
Should be just "addCycle", right?

Reply:
Not every cycle in a graph is a Hamiltonian cycle.

Reply:
Or do you mean the case that this helper method is called upon a non-empty graph? In that case you would be right, the cycle added would not be Hamiltonian because it only spans the newly added vertices, not possible previously existing ones. I always call this method upon a newly created, empty graph. But that is not checked or enforced, so I guess renaming the method is okay.

Reply:
Yes, that's what I meant.